### ISSUE TITLE
Cannot find field in google::protobuf::util::JsonToBinaryString

### ISSUE THREAD

**Issue by SRyabinin** [google/protobuf#1415]
I get status.ok()==false and error: "Cannot find field".
JsonToBinaryString confuses two field with same name "Value" and cannot parse json created by BinaryToJsonString.

This is my proto:

```
syntax = "proto3";
import "google/protobuf/wrappers.proto";

message Test0{
    bytes data = 1;
}
message Test1{
    google.protobuf.Int32Value div  = 2;
}
message Envelop
{
    oneof message
    {
        Test0 Value = 20;
        Test1 Test1 = 21;
    }
}
```

[protobufFromJsonFailed.zip](https://github.com/google/protobuf/files/220640/protobufFromJsonFailed.zip)

**Comment by xfxyjwf** [comment]
Could you also post the JSON text that the parser failed to parse?

**Comment by SRyabinin** [comment]
The problem json is: {"test1":{"div":1}}".
The text of error looks useless: "Test1.div: Cannot find field.". You can see that Test1 has "div" field. If I rename "Test0 Value" to "Test0 Value1" the problem will disappear.

**Comment by SRyabinin** [comment]
Hi Feng Xiao, 
Does this task have a chance to be resolved before release?

**Comment by xfxyjwf** [comment]
Sorry, I hadn't looked into it yet. It will probably be included as a 3.0.x patch release.
